### ISSUE TITLE
Insert reference generates "Wrong type argument"

### ISSUE THREAD

**Issue by scottmartincampbell** [tmalsburg/helm-bibtex#31]
Hi,

I was trying the `helm-bibtex-insert-reference` command today using the most recent helm-bibtex on melpa and noticed that any entry with the author field containing "Lastname, Firstname" generated `wrong type argument`. Author fields containing "Firstname Lastname" were fine. I didn't try any with multiple authors. The error pops up within `helm-bibtex-apa-format-authors`. 

Thanks in advance, if you can get a chance to confirm and/or fix it.

Scott

**Comment by tmalsburg** [comment]
I can't replicate this.  Could you please show me the entry that's causing the problem?  Thank you.

**Comment by scottmartincampbell** [comment]
The following entry did not work for me:

```
@Manual{wetterlund2013_if,
	file = {~/edoc/wetterlund2013.pdf},
	timestamp = {2015-01-13 013d},
	author = {Wetterlund, Kris},
	title = {If you Can't See It Don't Say It},
	year = 2013,
	date = {2013-7-3},
	publisher = {Museum-Ed},
	url = {http://tamtamsp.pressbooks.com/},
	tags = {:to read},
	keywords = {*oma exhibits}
}
```

**Comment by tmalsburg** [comment]
Weird, it works in my setup.  Are you using the latest version from git master?

**Comment by scottmartincampbell** [comment]
I'm using  `helm-bibtex-20150129.2118` from melpa, so yes. Everything else is up to date as well. I'll see if I can isolate something and see if it's something in my setup somewhere. I tried just that one entry alone in a file and that didn't help. 

Here's the top of the debug, in the meantime:
```
Debugger entered--Lisp error: (wrong-type-argument sequencep 10)
  s-left(1 10)
  #[(it) "\301\302\"\303P\207" [--cl-var-- s-left 1 "."] 3]("Kris")
  mapcar(#[(it) "\301\302\"\303P\207" [--cl-var-- s-left 1 "."] 3] ("Kris"))
  helm-bibtex-apa-format-authors("Wetterlund, Kris")
  helm-bibtex-apa-get-value("author" (("keywords" . "{*oma exhibits}") ("tags" . "{:to read}") ("url" . "{http://tamtamsp.pressbooks.com/}") ("publisher" . "{Museum-Ed}") ("date" . "{2013-7-3}") ("year" . "2013") ("title" . "{If you Can't See It Don't Say It}") ("author" . "{Wetterlund, Kris}") ("timestamp" . "{2015-01-13 013d}") ("file" . "{~/edoc/wetterlund2013.pdf}") ("=type=" . "Manual") ("=key=" . "wetterlund2013_if")))
```

**Comment by tmalsburg** [comment]
Huh, this doesn't make any sense at all.  Where is the 10 in the `s-left` call coming from? It should be `(s-left 1 "Kris")` which gives us the first charakter of the first name.

**Comment by scottmartincampbell** [comment]
The 10 in `s-left` is coming from the position of the `,` on L522: `else if (s-index-of "," a)`. Longer or shorter lasnames change the position of the `,` and throw a different error. I tried `else if (s-contains? "," a)` and the error was `wrong-type-argument sequencep, t`. So, the 10 is get:ting collected in the `cl-loop` somehow, I guess, but my lisp is pretty weak so I'll need to dig further. I have updated `s.el` and `dash.el` libraries, and `cl-lib.el 0.5` installed, but maybe there's a clash with the emacs 24.3 `cl-lib`? I'm out of my depth, but will keep trying. Sorry for the mess.
